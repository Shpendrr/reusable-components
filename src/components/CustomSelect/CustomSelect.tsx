import React, { FC } from "react";

export interface IOption {
  value: string;
  label: string;
}

export interface ICustomSelect<T> {
  options: T;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
}

const CustomSelect: FC<ICustomSelect<IOption[]>> = ({
  options,
  value,
  onChange,
  ...rest
}) => {
  return (
    <select value={value} {...rest} onChange={onChange}>
      {options?.map((option: IOption) => (
        <option key={option.value} value={option.value}>
          {option.label}
        </option>
      ))}
    </select>
  );
};

export default CustomSelect;
