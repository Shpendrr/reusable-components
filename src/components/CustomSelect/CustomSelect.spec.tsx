import React from "react";
import { render } from "@testing-library/react";
import CustomSelect, {
  ICustomSelect,
  IOption,
} from "@/components/CustomSelect/CustomSelect";
import "@testing-library/jest-dom";

describe("CustomSelect", () => {
  const options: IOption[] = [
    { value: "option1", label: "Option 1" },
    { value: "option2", label: "Option 2" },
    { value: "option3", label: "Option 3" },
  ];

  it("renders options correctly", () => {
    const { getByText } = render(
      <CustomSelect options={options} value="option1" onChange={() => {}} />
    );
    expect(getByText("Option 1")).toBeInTheDocument();
    expect(getByText("Option 2")).toBeInTheDocument();
    expect(getByText("Option 3")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const customSelectProps: ICustomSelect<IOption[]> = {
      options,
      value: "option1",
      onChange: () => {},
    };
    const { container } = render(<CustomSelect {...customSelectProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
