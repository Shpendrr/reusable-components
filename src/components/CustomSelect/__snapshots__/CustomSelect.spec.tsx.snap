// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect matches snapshot 1`] = `
<select>
  <option
    value="option1"
  >
    Option 1
  </option>
  <option
    value="option2"
  >
    Option 2
  </option>
  <option
    value="option3"
  >
    Option 3
  </option>
</select>
`;
